lib/src/structs.dart

```diff
@@ -671,35 +671,17 @@ class ACCEL extends Struct {
 class MONITORINFO extends Struct {
   @Uint32()
   external int cbSize;
-  @Int32()
-  external int rcMonitorLeft;
-  @Int32()
-  external int rcMonitorTop;
-  @Int32()
-  external int rcMonitorRight;
-  @Int32()
-  external int rcMonitorBottom;
-  @Int32()
-  external int rcWorkLeft;
-  @Int32()
-  external int rcWorkTop;
-  @Int32()
-  external int rcWorkRight;
-  @Int32()
-  external int rcWorkBottom;
+
+  external RECT rcMonitor;
+  external RECT rcWork;
+
   @Uint32()
   external int dwFlags;
 
   factory MONITORINFO.allocate() => allocate<MONITORINFO>().ref
     ..cbSize = sizeOf<MONITORINFO>()
-    ..rcMonitorLeft = 0
-    ..rcMonitorTop = 0
-    ..rcMonitorRight = 0
-    ..rcMonitorBottom = 0
-    ..rcWorkLeft = 0
-    ..rcWorkTop = 0
-    ..rcWorkRight = 0
-    ..rcWorkBottom = 0
+    ..rcMonitor = RECT.allocate()
```

Comment:
@dcharkes What's the right way to initialize these variables now that they're nested structs? Is this going to work? Or do I have to go through every field in the nested struct and initialize them manually?

Reply:
`RECT.allocate()` would malloc it, copy by value, and discard the pointer leaking memory.

If you want to initialize everything to zero you might instead want to use `memset`, and wrap `allocate` so that it 0-initializes everything. `allocate<MONITORINFO>().ref` -> `allocate<MONITORINFO>()..memset(0, sizeOf<MONITORINFO>()).ref`

Reply:
`memset`! Why have I never seen this in our examples... that's really helpful. I'll update accordingly. Thank you!!!

Reply:
Wait -- there _is_ no `memset`! Are you thinking of the C `memset()` call, @dcharkes? Or is that in a different branch?

Reply:
Never mind. I realized that the `allocate()` doesn't do more than wrap the Win32 `HeapAlloc()` call, so I've replaced it with my own `zeroAllocate()` function that uses the `HEAP_ZERO_MEMORY` flag. I'm going to raise an issue suggesting we use this by default (along with `calloc` on the UNIX side).